### ISSUE TITLE
Working TypeScript example, can't get it to work

### ISSUE THREAD

**Issue by MarkusWendorf** [developit/unistore#119]
Good evening,

I'm trying to get unistore to work with TypeScript and properly type my store. I would like to ask for some help on my example (Typescript on React).

Repo: [Unistore TypeScript](https://github.com/MarkusWendorf/unistore-typescript)
I'm getting an error: TS2604: JSX element type 'App' does not have any construct or call signatures.
If I ignore the error with @ts-ignore everything works as expected.

What am I missing?

PS:
I would appreciate if you would include a working TypeScript example with every release.

**Comment by joseluisq** [comment]
Here a TypeScript (TSX) working example using v3.1.0 https://github.com/joseluisq/preact-starter

**Comment by stephan281094** [comment]
I'm having the same issue, using React. It seems as if `connect` returns an instance of the component, instead of the constructor. Have a look at the following code snippet.

```tsx
import * as React from "react";
import { render } from "react-dom";
import createStore from "unistore"
import { connect, Provider } from "unistore/react";

const store = createStore({ bar: "baz" });

const Foo = connect("bar")((props: any) => (
  <h1>{props.bar}</h1>
));

// Throws "JSX element type 'Foo' does not have any construct or call signatures. [2604]".
const App = () => (
  <Provider store={store}>
    <Foo />
  </Provider>
);

render(<App />, document.querySelector(".mount"));
```

**@types/react**: 16.7.7
**React**: 16.6.3
**Typescript**: 3.1.6
**Unistore**: 3.1.0

**Comment by joseluisq** [comment]
@stephan281094 No, I don't but did you tried [this one](https://github.com/GantMan/ReactStateMuseum/tree/master/React/unistore)?

**Comment by stephan281094** [comment]
@joseluisq I actually meant an example where they use both React and TypeScript. The example of ReactStateMuseum isn't written in TypeScript.

To clarify, I did manage to get unistore working with React in plain javascript, but not with TypeScript due to the error @MarkusWendorf is describing in this issue.

**Comment by joseluisq** [comment]
> @joseluisq I actually meant an example where they use both React and TypeScript. The example of ReactStateMuseum isn't written in TypeScript.
> 
> To clarify, I did manage to get unistore working with React in plain javascript, but not with TypeScript due to the error @MarkusWendorf is describing in this issue.

Sorry by my incomprehension.

**Comment by joseluisq** [comment]
Here my exploration about the current error:

**unistore/preact**
![screen shot 2018-11-27 at 15 44 26](https://user-images.githubusercontent.com/1700322/49089524-06374a80-f25c-11e8-8e2d-37750fdf93b5.png)

`Preact.ComponentConstructor` as result 

**unistore/react**
![screen shot 2018-11-27 at 15 42 55](https://user-images.githubusercontent.com/1700322/49089441-dab46000-f25b-11e8-9524-2bacc100b233.png)

`React.Component` as result

**unistore/react**
![screen shot 2018-11-27 at 15 46 43](https://user-images.githubusercontent.com/1700322/49089825-aab98c80-f25c-11e8-8db1-9ec9d8b856d1.png)

**Note:** Just changing `React.Component` to `React.ComponentClass` solves the issue.

Thoughts?

**Comment by stephan281094** [comment]
Thanks for the quick response. Your change got rid of the error, great work! Not sure if this change is compatible with previous versions of React though. Might require some research.

**Comment by developit** [comment]
looks like `React.ComponentClass` has been available since React 15.

**Comment by developit** [comment]
I've opened a PR with the above change.

**Comment by stephan281094** [comment]
Awesome, thanks!

**Comment by developit** [comment]
Released as [3.2.0](https://github.com/developit/unistore/releases/tag/3.2.0)!

**Comment by joseluisq** [comment]
:+1: 

@developit Is #127 issue already fixed too?
